src/overlays/actors/ovl_En_Mu/z_en_mu.c

```diff
@@ -44,27 +34,179 @@ static ColliderCylinderInit D_80AB0BD0 = {
+            EnMu_Interact(thisx, globalCtx);
+            return 0;
+        default:
+            return 1;
+    }
+}
+
+void EnMu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    ColliderCylinder* colliderCylinder;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 160.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06004F70, &D_060003F4, 0, 0, 0);
+    colliderCylinder = &this->collider;
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, (Actor*)this, &D_80AB0BD0);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &D_80AB0BFC);
+    this->actor.targetMode = 6;
+    Actor_SetScale((Actor*)this, 0.01f);
+    EnMu_Interact(this, globalCtx);
+    EnMu_SetActionFunction(this, EnMu_Pose);
+}
+
+void EnMu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void EnMu_Pose(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    func_80034F54(globalCtx, &this->jointTable, &this->morphTable, 0x10);
+}
+
+void EnMu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    s32 pad;
+    f32 talkDist;
+    Vec3s pos;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y;
+    pos.z = this->actor.world.pos.z;
+
+    this->collider.dim.pos = pos;
+
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(this, globalCtx);
+    talkDist = this->collider.dim.radius + 30.0f;
+    func_800343CC(globalCtx, &this->actor, this->dialogState, talkDist, EnMu_GetFaceReaction, EnMu_CheckDialogState);
+
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.focus.pos.y += 60.0f;
+}
+
+s32 EnMu_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, s32 arg2, s32 arg3, Vec3s* rot, void* thisx) {
+    EnMu* this = THIS;
+    if ((limbIndex == 5) || (limbIndex == 6) || (limbIndex == 7) || (limbIndex == 0xB) || (limbIndex == 0xC) ||
+        (limbIndex == 0xD) || (limbIndex == 0xE)) {
+        rot->y += Math_SinS(this->jointTable[limbIndex]) * 200.0f;
+        rot->z += Math_CosS(this->morphTable[limbIndex]) * 200.0f;
+    }
+    return 0;
+}
+
+void EnMu_PostLimbDraw(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+}
+
+Gfx* EnMu_DisplayListSetColor(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b, u8 a) {
+    Gfx* dlist;
+    dlist = Graph_Alloc(gfxCtx, 2 * sizeof(Gfx));
+    gDPSetEnvColor(dlist, r, g, b, a);
+    gSPEndDisplayList(dlist + 1);
+    return dlist;
+}
+
+void EnMu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+
+    // 2 sets of 5 colors for each actor in town with different colors
+    Color_RGBA8 colors[2][5] = { 0x64, 0x82, 0xEB, 0x00, 0xA0, 0xFA, 0x3C, 0x00, 0x5A, 0x3C, 0x14, 0x00, 0x1E, 0xF0,
```

Comment:
Is it possible to use all the layers of braces for these, to show all the separate RGBAs and arrays?

Reply:
additionally, colors should be represented in decimal.